src/server.cpp

```diff
@@ -3510,10 +3510,15 @@ v3f Server::findSpawnPos()
 	}
 
 	bool is_good = false;
+	// Calculate mapgen edges
+	map.getMapgenParams()->calcMapgenEdges();
+	// Limit spawn range to mapgen edges (determined by 'mapgen_limit')
+	s32 range_max = MYMIN(-map.getMapgenParams()->mapgen_edge_min,
```

Comment:
what i suggest is to create a public function on mapgenparams s32 getRangeMax() which does what this line does + calcMapGenParams (which should remain private then) and call it here